File: Net7 - Copy/020-039/038 CS Tableaux/MyApp.cs
// 38 C# tableaux
// Essais sur la création dynamique de tableaux
//
// 2001-02-08   PV
// 2006-10-01   PV  VS2005
// 2012-02-25   PV  VS2010
// 2021-09-18   PV  VS2022, Net6

using System;
using static System.Console;

internal static class MyApp
{
    public static void Main()
    {
        var t1 = new int[10];

        int[] t2 =
        {
            1,2,3,4,5
        };

        var t3 = new int[]
        {
          1,2,3,4,5
        };

        unchecked
        {
            var a = 500000;
            var b = 400000;
            t1[0] = a * b;
        }
        WriteLine("t1[0]: {0}", t1[0]);

        object[] u1;
        u1 = null;
        WriteLine("u1==null: {0}", u1 == null);

        u1 = new object[10];
        WriteLine("u1[0]==null: {0}", u1[0] == null);

        u1[0] = new object();
        WriteLine("u1[0]==null: {0}", u1[0] == null);

        _ = Console.ReadLine();
    }
}

File: Net8/001-019/009 VB Inheritance/MyApp.vb
' 09 VB héritage
' Essais d'héritage de classes en VB
'
' 2001-10-27    PV
' 2006-10-01 	PV		VS2005
' 2012-02-25	PV		VS2010
' 2021-09-17 	PV		VS2022/Net6
' 2023-01-10	PV		Net7

Module Module1

    Sub Main()
        Dim d As New D()
        Console.ReadLine()
    End Sub

End Module

Class B

    Sub New()
        WriteLine("B.ctor")
    End Sub

End Class

Class D
    Inherits B

    Sub New()
        'MyBase.New()   'Inutile, appelé automatiquement
        WriteLine("D.ctor")
    End Sub

End Class
